util.go

```diff
@@ -154,11 +155,17 @@ func trimeSchemaRefLinkPrefix(ref string) string {
 	return strings.TrimPrefix(ref, "#/components/schemas/")
 }
 
-func genSchemeaObjectID(pkgName, typeName string) string {
-	typeNameParts := strings.Split(typeName, ".")
-	pkgName = replaceBackslash(pkgName)
+func genSchemeaObjectID(pkgName, typeName string, p *parser) string {
+	validatedPkgName := checkAndMutatePackageName(pkgName, p)
+	validatedTypeName := checkAndMutateTypeName(typeName, p)
```

Comment:
This feels more intrusive than it needs to be. Can we use the unmodified id throughout the code and only swap in the aliases when writing the spec?